### PR TITLE
feat: added lead appender task

### PR DESCRIPTION
**Pull request by arunp7**

### CODE REVIEW

File: NAVIGATOR_LEAD_APPENDER.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    
+    // Fetch and Macth through List Name
+    await page.waitForSelector('[data-control-name="view_lead_lists_from_app_header"]')
+    await page.click('[data-control-name="view_lead_lists_from_app_header"]')
+    await page.waitForSelector('.artdeco-models-table-row.ember-view')
+    let cliked = await page.evaluate((listname)=>{
+        let list_names = document.querySelectorAll('[data-control-name="view_list_detail"]')
+        for(let i = 0 ; i<=list_names.length; i++){
+        if(list_names[i].innerText.toLocaleLowerCase().includes(listname.toLocaleLowerCase())){
+            console.log('1')
```
Comment (theproductiveprogrammer) [inline NAVIGATOR_LEAD_APPENDER.js:56]: Missed in your clean up (console.log(1))

File: NAVIGATOR_LEAD_APPENDER.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        let i_date = new Date(task.inputdate)
+        let leads = []
+        let l = 0
+        do{
+        await page.waitForSelector('[aria-label="Lead Name"]')
+        await page.waitForSelector('[data-control-name="view_list_detail_lead_panel_via_lead_name"]')
+        await page.waitForSelector('.artdeco-models-table-cell.list-people-detail-header__last-updated.ember-view')
+        await page.waitForSelector(".artdeco-models-table-header.artdeco-models-table-header.list-people-detail-header__last-updated.ember-view")
+        const lead_ele = await page.$$eval('[data-control-name="view_list_detail_lead_panel_via_lead_name"]', 
+        names => names.map(name => name.innerText))
```
Comment (theproductiveprogrammer) [inline NAVIGATOR_LEAD_APPENDER.js:75]: Nice technique

File: NAVIGATOR_LEAD_APPENDER.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        const dates  = await page.$$eval('.artdeco-models-table-cell.list-people-detail-header__last-updated.ember-view', 
+        dates =>  dates.map(date => date.innerText)
+            );
+        let url_elem = await page.$$('[data-control-name="view_list_detail_lead_panel_via_lead_name"]')
+        if(url_elem){
+        for(let j = 0 ; j<lead_ele.length; j++){
+            let name = lead_ele[j]
+            let updtd_date_ele = dates[j]
+            let updtd_date = new Date(updtd_date_ele)
```
Comment (theproductiveprogrammer) [inline NAVIGATOR_LEAD_APPENDER.js:84]: Are we assuming here that `new Date()` can parse all date types? Would suggest using a library like `day.js` and several defined formats

File: NAVIGATOR_LEAD_APPENDER.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+            if(updtd_date >= i_date)
```
Comment (theproductiveprogrammer) [inline NAVIGATOR_LEAD_APPENDER.js:85]: This is a tricky one - what about leads added on the same date? How do we not duplicate them?

**Review by theproductiveprogrammer — CHANGES_REQUESTED** [review]

File: NAVIGATOR_LEAD_APPENDER.js (hunk 1 above)
Comment (arunp7) [reply NAVIGATOR_LEAD_APPENDER.js:56]: oh ! Will correct Charles.

File: NAVIGATOR_LEAD_APPENDER.js (hunk 2 above)
Comment (arunp7) [reply NAVIGATOR_LEAD_APPENDER.js:75]: Thanks Charles.

File: NAVIGATOR_LEAD_APPENDER.js (hunk 4 above)
Comment (arunp7) [reply NAVIGATOR_LEAD_APPENDER.js:85]: I thought about that, we will have to add some kind of logic in the backend (linkedin url is only the unique thing that we could check for duplicate with the data we have) or else we will have to add some input variables

**Review by theproductiveprogrammer — APPROVED** [review]

**Pull request merged**